cep-XXXX.md

```diff
@@ -0,0 +1,216 @@
+- `./deactivate.d/`: Directory containing shell scripts. They SHOULD be executed on environment deactivation, in reverse alphabetical order.
+
+### Pre- and post-link/unlink scripts
+
+The `bin/` and `Scripts/` directories usually contain executables populated by the packages themselves. There are four special paths that SHOULD be handled by conda clients. For files named like `.{package-name}-{action}.{extension}`, where `{package-name}` corresponds to the package name and `{extension}` is either `sh` (Unix) or `bat` (Windows), there are four possible `{action}` values and associated behaviors:
+
+- `pre-link`: Executed before a package is installed / linked. Deprecated; conda clients SHOULD ignore them.
+- `post-link`: Executed after a package is installed / linked.
+- `pre-unlink`: Executed before a package is removed / unlinked.
+- `post-unlink`: Executed after a package is removed / unlinked. Deprecated; conda clients SHOULD ignore them.
+
+Execution is performed in topological order. The conda clients SHOULD export these environment variables for the scripts to use:
+
+- `ROOT_PREFIX`: Path to the conda client installation root, when applicable.
+- `PREFIX`: Path to the environment where the package is installed.
+- `PKG_NAME`: Name of the package.
+- `PKG_VERSION`: Version of the package.
+- `PKG_BUILDNUM`: Build number of the package.
+
+### Top-level `condarc` files
+
+This can be used to configure behavior of the conda client performing operations on this environment. The following locations are recognized as valid configuration sources.
+
+- `./.condarc`
+- `./condarc`
+- `./condarc.d/*.yml`
+- `./condarc.d/*.yaml`
+
+## Management of a conda environment
+
+### Creating a conda environment
+
+An empty directory `$CONDA_PREFIX` can be turned into a conda environment by creating an empty `conda-meta/history` file. The conda client MAY register this location into a central registry of environments, such as `~/.conda/environments.txt`.
+
+The command used to create that environment MAY be recorded in `conda-meta/history`, along with the version, timestamp and details of the transaction.
+
+### Installing packages
+
+For each package to be installed, assuming it has been already downloaded and extracted:
+
+- For regular (non-`noarch: python`) packages, place the contents of the artifact into `$CONDA_PREFIX`.
+  - If the file contains a prefix placeholder, replace it with the value of `$CONDA_PREFIX` and copy the file.
+  - Otherwise, link into `$CONDA_PREFIX` (or copy if linking is not possible).
+- For `noarch: python` packages, the contents of the artifact MUST be linked or copied under `$CONDA_PREFIX/lib/pythonX.Y` (Unix, where `X.Y` represents the MAJOR.MINOR Python version) or `$CONDA_PREFIX/Lib` (Windows). If the package contains `entry_points`, the necessary scripts MUST be generated under `$CONDA_PREFIX/bin` (Unix) or `$CONDA_PREFIX/Scripts` (Windows).
```

Comment:
For `noarch: python` packages only the files in the `site-packages` directory are linked/copied into the site-packages directory. Other files are linked/copies into the prefix in the location they appear in the package.

Also CEP-19 comes into play for the location of the site-packages directory. It is not always lib/pythonX.Y/site-packages (unix) or Lib/site-packages (Windows)

Reply:
Changed this part to refer to CEP 20.